src/cast_core/src/CrudActor.cpp

```diff
@@ -1164,14 +1247,20 @@ struct CrudActor::CollectionName {
 
 struct CrudActor::PhaseConfig {
     std::vector<std::unique_ptr<BaseOperation>> operations;
+    std::vector<std::unique_ptr<State>> states;
+    std::vector<double> initial_state_weights;
     metrics::Operation metrics;
     std::string dbName;
     CrudActor::CollectionName collectionName;
+    bool continue_current_state;
+
+    // min time of next operation
 
     PhaseConfig(PhaseContext& phaseContext, mongocxx::pool::entry& client, ActorId id)
         : dbName{getDbName(phaseContext)},
           metrics{phaseContext.actor().operation("Crud", id)},
-          collectionName{phaseContext} {
+          collectionName{phaseContext},
+          continue_current_state{phaseContext["Continue"].maybe<bool>().value_or(false)} {
```

Comment:
The `PhaseContext` has access to the parent `ActorContext` which can then iterate over all the `Phase`s. That would be a good thing to add before this is used in production since new users are likely to mis-configure things the first time using it.